src/peergos/server/simulation/AccessControl.java

```diff
@@ -94,9 +94,7 @@ private AccessControlUnit getAcu(FileSystem.Permission permission) {
                 case WRITE:
                     return writers.getAllowed(path);
                 case READ:
-                    List<String> allowed = readers.getAllowed(path);
-                    List<String> allowed2 = writers.getAllowed(path);
-                    return new ArrayList<>(Stream.of(allowed, allowed2).flatMap(e -> e.stream()).collect(Collectors.toSet()));
+                    return readers.getAllowed(path);
```

Comment:
Can you explain this change? This was done because write access implies read access in cryptree.

Reply:
It still does. When verifying access permissions in verifySharingPermissions() it takes that into account.

Reply:
I'm not seeing where that happens. Where do you mean?

Reply:
Simulator.java line 506: case WRITE:  You will see that for write permission is first checks fs.read(path); and then fs.write(path, new byte[]{read[0]});